### ISSUE TITLE
When uploading large files which takes more then 100 seconds to complete, I get Exception

### ISSUE THREAD

**Issue by StefH** [skazantsev/WebDavClient#3]
When using PutFile(. . . ) and the upload takes more then 100 seconds to upload, I get this exception

```
InnerException = {System.ObjectDisposedException: Cannot access a closed file.
   at System.IO.__Error.FileNotOpen()
   at System.IO.FileStream.get_Position()
   at StackSharing.Lib.StackClient.UploadFile(UploadStatus status, OnlineItem folder, String filePath)
```

It's probably because when creating a HttpClient, there is no possible override from the default timeout from 100 seconds:

https://github.com/skazantsev/WebDavClient/blob/master/src/WebDav.Client/WebDavClient.cs#L711

**Comment by skazantsev** [comment]
@StefH Thanks for bringing this up. The issue is fixed in the version 1.0.3, a timeout can be set when creating an instance of WebDavClient.

**Comment by StefH** [comment]
I'm not very sure it's the best solution to set this to infinite by default. Maybe make timeout a nullable int, or set it only to the webclient if specified.

Because users from this lib would maybe still expect 100 when nothing is set.
